src/wobbly-effect.cpp

```diff
@@ -327,6 +327,33 @@ endless_shell_fx_wobbly_move_by (EndlessShellFXWobbly *effect,
     }
 }
 
+#define PAINT_BOX_OFFSET 1
+
+static gboolean
+endless_shell_fx_get_untransformed_paint_box (ClutterActor    *actor,
+                                              ClutterActorBox *box)
+{
+  /* Get the actor's parent class paint volume, bypassing affine
+   * transformations (except translation). */
+  const ClutterPaintVolume *volume = clutter_actor_get_paint_volume (actor);
+  ClutterVertex origin;
+
+  if (volume == NULL)
+    return FALSE;
+
+  /* We don't have access to the stage projection matrix
+   * so the best we can do is hope here that the volume is
+   * two dimensional and orthogonal. */
```

Comment:
I believe that we do... see `clutter_actor_get_transform()` and `clutter_paint_volume_get_depth()`

Reply:
Hmm, I guess I meant that in the sense that the actor needs to be a camera-facing plane for this to work. If it was three-dimensional this probably wouldn't work - in that case we would use the modelview matrix and premultiply it by the projection matrix, then postmultiply each of its vertices, but we don't have access to the projection matrix here.